### PR TITLE
check for openshift /oapi endpoint

### PR DESCRIPTION
**Pull request by sub-mod**
check for openshift /oapi endpoint if available set a flag in client

### CODE REVIEW

**Comment by stefwalter** [comment]
Hmmm, so openshift serves all its objects under ```/oapi```? I was under the impression the objects were available in /api. If not, this really breaks #2463 ... as it only proxies ```/api```

**Comment by sub-mod** [comment]
@stefwalter  The OpenShift distribution of Kubernetes includes the Kubernetes v1 REST API and the OpenShift v1 REST API. 
all openshift objects not in k8 are available in /oapi
and all k8 objects are at /api
https://docs.openshift.com/enterprise/3.0/rest_api/openshift_v1.html

**Comment by stefwalter** [comment]
Good to know. So I guess this conflicts somewhat with #2463. We'll probably have to use 'oc proxy' opportunistically as well as 'kubectl proxy' in order to access both kinds of objects.

**Comment by stefwalter** [comment]
The RHEL testing problem has been worked around here: #2613

**Comment by stefwalter** [comment]
Looks like test-kubernetes.html is failing. Could you fix that before review? Like so:

```
$ make
$ ./test-server &
$ firefox http://localhost:8765/pkg/kubernetes/test-kubernetes.html
```

**Comment by stefwalter** [comment]
This iterates through all the 'wrong' fallback schemes first before trying /api. I think that's a problem, and will slow down connecting. In addition if something else is listening on those ports it will be confusing.

Should we just try to use stuff /oapi and if it works, then we set the openshift available flag? Or will that happen too late to use it in UI decisions?

File: pkg/kubernetes/client.js [hunk 1]
```diff
@@ -505,7 +528,7 @@ define([
                 debug("kube config scheme:", scheme);
                 schemes.unshift(scheme);
             })
-            .always(step);
+            .always(step("/oapi"));
```
Comment (stefwalter) [inline pkg/kubernetes/client.js]: This runst the step function here once while registering the undefined return value from step("/oapi") as a callback. This seems like a bug.

**Comment by stefwalter** [comment]
Replaced by #2653 to fix review items.

**Pull request closed without merging**